Source/Particles/Pusher/UpdatePosition.H

```diff
@@ -15,6 +15,34 @@
 #include <AMReX_REAL.H>
 
 
+/** \brief Push the particle's positions over one timestep,
+ *    given the value of its momenta `ux`, `uy`, `uz` */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void UpdatePosition(amrex::Real& x, amrex::Real& y, amrex::Real& z,
+                    const amrex::Real ux, const amrex::Real uy, const amrex::Real uz,
+                    const amrex::Real dt )
```

Comment:
Is there a reason this isn't *always* `ParticleReal`, rather than needing both `Real` and `ParticleReal` versions?